--- cereal-1.1.2/include/cereal/details/polymorphic_impl.hpp	2015-06-27 16:55:15.191329000 -0400
+++ cereal/include/cereal/details/polymorphic_impl.hpp	2015-06-27 16:56:14.731329000 -0400
@@ -56,17 +56,17 @@
     have been registered with CEREAL_REGISTER_ARCHIVE.  This must be called
     after all archives are registered (usually after the archives themselves
     have been included). */
-#define CEREAL_BIND_TO_ARCHIVES(T)                   \
+#define CEREAL_BIND_TO_ARCHIVES(...)                   \
     namespace cereal {                               \
     namespace detail {                               \
     template<>                                       \
-    struct init_binding<T> {                         \
-        static bind_to_archives<T> const & b;        \
+    struct init_binding<__VA_ARGS__> {                         \
+        static bind_to_archives<__VA_ARGS__> const & b;        \
         static void unused() { (void)b; }            \
     };                                               \
-    bind_to_archives<T> const & init_binding<T>::b = \
+    bind_to_archives<__VA_ARGS__> const & init_binding<__VA_ARGS__>::b = \
         ::cereal::detail::StaticObject<              \
-            bind_to_archives<T>                      \
+            bind_to_archives<__VA_ARGS__>                      \
         >::getInstance().bind();                     \
     }} /* end namespaces */
 
